Prepare PyPi distribution for release 0.3.0

ckadner opened the pull request:
**Description of your changes:**

Documentation update for the 0.3.0 release on PyPi.

Preview release: https://test.pypi.org/project/kfp-tekton/0.3.0rc5/

**Environment tested:**

* Python Version (use `python --version`): 3.7.5
* Tekton Version (use `tkn version`): 0.15
* Kubernetes Version (use `kubectl version`): 1.16
* OS (e.g. from `/etc/os-release`): Mac OS 10.15

/cc @animeshsingh

ckadner commented:
@Tomcli -- I removed the "Optional Features" section and moved the "Building Tekton from Master" paragraph to the `FEATURES.md` doc just as an informational section since we don't currently rely on Tekton master for any of our features

Tomcli commented:
> @Tomcli -- I removed the "Optional Features" section and moved the "Building Tekton from Master" paragraph to the `FEATURES.md` doc just as an informational section since we don't currently rely on Tekton master for any of our features

Can we not put it under FEATURES? For 0.3 release we don't rely anything on master, so it could confuse people who are new to this repo. Can we put it under developer guide instead?

ckadner commented:
> > @Tomcli -- I removed the "Optional Features" section and moved the "Building Tekton from Master" paragraph to the `FEATURES.md` doc just as an informational section since we don't currently rely on Tekton master for any of our features
> 
> Can we not put it under FEATURES? For 0.3 release we don't rely anything on master, so it could confuse people who are new to this repo. Can we put it under developer guide instead?

Sure. Done.

Tomcli commented:
/lgtm

ckadner commented:
@animeshsingh -- If you are okay with the changes to the SDK README and how it shows up on Test-PyPI (https://test.pypi.org/project/kfp-tekton/0.3.0rc5/#description) then I will upload the `0.3.0` release to the real Python Package Index

animeshsingh commented on sdk/README.md:
```diff
@@ -133,86 +130,21 @@ in the same directory:
-cluster with `kubectl`. The Tekton server will automatically start a pipeline run
-for which we can follow the logs using the `tkn` CLI. 
+cluster with `kubectl`. The Tekton server will automatically start a pipeline run.
+We can follow the logs using the `tkn` CLI.
```
```suggestion
We can then follow the logs using the `tkn` CLI.
```

animeshsingh commented on sdk/README.md:
```diff
@@ -133,86 +130,21 @@ in the same directory:
-
-## Optional Features
-
-### Compile Kubeflow Pipelines without Artifacts
```
We are removing this because this moved to server side 
cc @Tomcli

Tomcli replied:
yes, I discussed with Christian and this is no longer needed (for SDK).

ckadner replied:
we originally had this section for:
- generate optional PipelineRun ... no longer optional
- enable (later disable) artifacts ... server side now

animeshsingh commented:
/lgtm
/approve

ckadner commented:
Thanks @Tomcli @animeshsingh 

The v`0.3.0` on **PyPI**: https://pypi.org/project/kfp-tekton/0.3.0/

One-line install: `pip install kfp-tekton`
